qiskit/algorithms/evolvers/pvqd/pvqd.py

```diff
@@ -0,0 +1,364 @@
+
+"""The projected Variational Quantum Dynamics Algorithm."""
+
+from typing import Optional, Union, List, Tuple, Callable
+
+import logging
+import numpy as np
+
+from qiskit import QiskitError
+from qiskit.algorithms.optimizers import Optimizer, Minimizer
+from qiskit.circuit import QuantumCircuit, ParameterVector
+from qiskit.circuit.library import PauliEvolutionGate
+from qiskit.providers import Backend
+from qiskit.opflow import (
+    OperatorBase,
+    CircuitSampler,
+    ExpectationBase,
+    StateFn,
+)
+from qiskit.synthesis import EvolutionSynthesis, LieTrotter
+from qiskit.utils import QuantumInstance
+
+from .pvqd_result import PVQDResult
+from .utils import _get_observable_evaluator, _is_gradient_supported
+
+from ..evolution_problem import EvolutionProblem
+from ..evolution_result import EvolutionResult
+from ..real_evolver import RealEvolver
+
+logger = logging.getLogger(__name__)
+
+
+class PVQD(RealEvolver):
+    """The projected Variational Quantum Dynamics Algorithm.
+
+    In each timestep this algorithm computes the next state with a Trotter formula
+    (specified by the ``evolution`` argument) and projects it onto a variational form (``ansatz``).
+    The projection is determined by maximizing the fidelity of the Trotter-evolved state
+    and the ansatz, using a classical optimization routine. See Ref. [1] for details.
+
+    Example:
+
+        .. code-block:: python
+
+            import numpy as np
+
+            from qiskit import BasicAer
+            from qiskit.circuit.library import EfficientSU2
+            from qiskit.opflow import X, Z, I, MatrixExpectation
+
+            backend = BasicAer.get_backend("statevector_simulator")
+            expectation = MatrixExpectation()
+            hamiltonian = 0.1 * (Z ^ Z) + (I ^ X) + (X ^ I)
+            observable = Z ^ Z
+            ansatz = EfficientSU2(2, reps=1)
+            initial_parameters = np.zeros(ansatz.num_parameters)
+
+            time = 0.02
+            optimizer = L_BFGS_B()
+
+            # setup the algorithm
+            pvqd = PVQD(
+                ansatz,
+                initial_parameters,
+                timestep=0.01,
+                optimizer=optimizer,
+                quantum_instance=backend,
+                expectation=expectation
+            )
+
+            # specify the evolution problem
+            problem = EvolutionProblem(
+                hamiltonian, time, aux_operators=[hamiltonian, observable]
+            )
+
+            # and evolve!
+            result = pvqd.evolve(problem)
+
+    References:
+
+        [1] Stefano Barison, Filippo Vicentini, and Giuseppe Carleo (2021), An efficient
+            quantum algorithm for the time evolution of parameterized circuits,
+            `Quantum 5, 512 <https://quantum-journal.org/papers/q-2021-07-28-512/>`_.
+    """
+
+    def __init__(
+        self,
+        ansatz: Optional[QuantumCircuit] = None,
+        initial_parameters: Optional[np.ndarray] = None,
```

Comment:
Do the initial parameters bound to the ansatz define the initial state? If so can we explicitly state this somewhere?

Reply:
Yes they do, I'll add a comment 👍🏻